llvm/lib/ProfileData/PGOCtxProfWriter.cpp

```diff
@@ -265,7 +275,16 @@ Error llvm::createCtxProfFromYAML(StringRef Profile, raw_ostream &Out) {
       if (!TopList)
         return createStringError(
             "Unexpected error converting internal structure to ctx profile");
-      Writer.writeContextual(*TopList, DC.TotalRootEntryCount);
+
+      ctx_profile::ContextNode *FirstUnhandled = nullptr;
+      for (const auto &U : DC.Unhandled) {
+        SerializableCtxRepresentation Unhandled;
+        Unhandled.Guid = U.first;
+        Unhandled.Counters.insert(Unhandled.Counters.begin(), U.second.begin(),
```

Comment:
Repeatedly inserting in the front is inefficient. How about reserve(size), append and then reverse the vector?

Reply:
wdym. this copies the counter values from `U.second` to `Unhandled`. `insert` with a first/last iterator pair will - afaik - allocate the diff (last - first) (caveats about difference being computable, but the source is also a vector). Also why would I reverse?

There *is* a more efficient alternative - having a `createNode` that takes the guid and the counters separately - but not quite worth it for the yaml converter, it's a test utility.

Reply:
Since they are both vectors, the allocation is optional (depending on the capacity of  Unhandled.Counters) but there will always be a copy to shift the original elements to the right. This is what I was referring to as inefficient. But since you mention this is a utility for a test, it doesn't matter much.